### DISCUSSION TITLE
Microphone component

### DISCUSSION THREAD

**Discussion by Fariba-86** [jspsych/jsPsych#1299] [Q&A]
Dear all,
I want to program my study online using jspsych, but I don't know if there is any microphone plugin defined to let me add the component. I couldn't find component or plugin of this type in jspsych.org. I wish to know how could I give it a start and help it run! I want to run it on cognition.run. 
Needed information may be that I have a set of images that appear on the screen one after the other each preceded by a word prime, then the pics are going to be named and the mic component helps with recording the response.
I also would like to know if this mic component records response time (RTs) in milliseconds?
Thanks for your help in advance.

**Comment by becky-gilbert** [comment]
Hi @Fariba-86, check out the *-audio-response plugins here: https://github.com/becky-gilbert/jsPsych/tree/audio-response/plugins
The plugin that has been used and tested the most is the image-audio-response plugin, which displays an image and begins recording the spoken response: https://github.com/becky-gilbert/jsPsych/blob/audio-response/docs/plugins/jspsych-image-audio-response.md
But there are also versions for presenting an audio stimulus (audio-audio-response) and an HTML stimulus (html-audio-response).

I hope this helps. Let me know if you have any problems using these plugins. They're still in development and we'd welcome any feedback.

**Reply by Fariba-86** [reply]
Hi @becky-gilbert. Thanks for your response. I'm giving it a try and let you know as soon as I could run it. There is one point. If I'm running the experiment on cognition.run, how should path be given to save the audio file?
The example code says: 
'function save_file_to_jatos(data)'
and also: 'jatos.uploadResultFile(blob, filename);' 
and: 'postprocessing: save_file_to_jatos,'
How to replace them with the path for 'cognition.run'?

Thanks in advance

**Reply by becky-gilbert** [reply]
Hi @Fariba-86, sorry for the very slow response! The code that you're referring to is an example of how to save a file containing the audio response data directly to a server. You could ask @javidalpe whether cognition.run currently does (or could in the future) support saving audio recordings, but I'm guessing not? Ifnot, your other option is to use the default `postprocessing` function, which saves the audio response as a base 64 encoded string. This string is then saved in your jsPsych data, and you can convert it back into audio offline. Keep in mind that encoding the audio response as a base 64 string will produce very large data files, so you may want to be careful about the number and durations of your recordings if you save the data this way. You can read more about these issues on my [documentation page for the image-audio-response plugin](https://github.com/becky-gilbert/jsPsych/blob/audio-response/docs/plugins/jspsych-image-audio-response.md).

**Reply by javidalpe** [reply]
Hi @Fariba-86, it's Javier from cognition.run

We don't have an API or library for storing files, so we cannot save the audio file from the experiment. But this seems like a useful feature so we will study implementing it. On the meantime you can try `postprocessing` as suggested by @becky-gilbert.
